tests/test_immutability_options.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
```

Comment:
Add a small comment, that this is an integration test, and it will not be executed by pytest.